godump.go

```diff
@@ -1235,3 +1236,61 @@ func (d *Dumper) redactedValue(v reflect.Value) string {
 	typeStr := d.getTypeString(v.Type())
 	return d.colorize(colorRed, "<redacted>") + d.colorize(colorGray, " #"+typeStr)
 }
+
+// isComplexValue reports whether v unwraps to a struct/map/slice/array.
+func isComplexValue(v reflect.Value) bool {
+	_, ok := complexBaseKind(v)
+	return ok
+}
+
+// complexBaseKind unwraps interfaces/pointers, rejects nil, and returns the underlying complex kind if present.
+func complexBaseKind(v reflect.Value) (reflect.Kind, bool) {
+	if !v.IsValid() {
+		return 0, false
+	}
+
+	for {
+		switch v.Kind() {
+		case reflect.Interface:
+			if v.IsNil() {
+				return 0, false
+			}
+			v = v.Elem()
+		case reflect.Ptr:
+			if v.IsNil() {
+				return 0, false
+			}
+			v = v.Elem()
+		default:
+			switch v.Kind() {
+			case reflect.Struct, reflect.Map, reflect.Slice, reflect.Array:
+				return v.Kind(), true
+			default:
+				return 0, false
+			}
+		}
+	}
```

Comment:
Any reason to do not do this ?


```suggestion
	for {
		switch v.Kind() {
		case reflect.Interface:
			if v.IsNil() {
				return 0, false
			}
			v = v.Elem()
		case reflect.Ptr:
			if v.IsNil() {
				return 0, false
			}
			v = v.Elem()
		case reflect.Struct, reflect.Map, reflect.Slice, reflect.Array:
			return v.Kind(), true
		default:
			return 0, false
		}
	}
```